pkg/finder/podresourcefinder.go

```diff
@@ -0,0 +1,480 @@
+// 	if err != nil {
+// 		log.Printf("Error in  ContStatusResp: %v", err)
+// 		return nil, err
+// 	}
+// 	return ContStatusResp, nil
+// }
+
+func (cpf *PodResourceFinder) updateNUMAMap(numaData map[int]map[v1.ResourceName]int64, ri ResourceInfo) {
+	if ri.Name == v1.ResourceCPU {
+		for _, cpuIDStr := range ri.Data {
+			cpuID, err := strconv.Atoi(cpuIDStr)
+			if err != nil {
+				log.Printf("cannot convert cpuID: %q", cpuIDStr)
+				continue
+			}
+			nodeNum, ok := cpf.cpuID2NUMAID[cpuID]
+			if !ok {
+				log.Printf("unknown cpuID: %d", cpuID)
+				continue
+			}
+			numaData[nodeNum][ri.Name]--
+		}
+		return
+	}
+	for _, pciAddr := range ri.Data {
+		nodeNum, ok := cpf.pciAddr2NUMAID[pciAddr]
+		if !ok {
+			log.Printf("unknown PCI address: %q", pciAddr)
+			continue
+		}
+		numaData[nodeNum][ri.Name]--
+	}
+}
+//
+// func (cpf *criFinder) listPodSandBoxResponse() (*pb.ListPodSandboxResponse, error) {
+// 	//ListPodSandbox
+// 	podState := &pb.PodSandboxStateValue{}
+// 	podState.State = pb.PodSandboxState_SANDBOX_READY
+// 	filter := &pb.PodSandboxFilter{}
+// 	filter.State = podState
+// 	request := &pb.ListPodSandboxRequest{
+// 		Filter: filter,
+// 	}
+// 	PodSbResponse, err := cpf.client.ListPodSandbox(context.Background(), request)
+// 	if err != nil {
+// 		fmt.Errorf("Error in listing ListPodSandbox : %v", err)
+// 		return nil, err
+// 	}
+// 	return PodSbResponse, nil
+// }
+//
+func (f *PodResourceFinder) isWatchable(podNamespace string) bool {
+	if f.args.Namespace == "" {
+		return true
+	}
+	//TODO:  add an explicit check for guaranteed pods
+	return f.args.Namespace == podNamespace
+}
+
+func (f *PodResourceFinder) Scan() ([]PodResources, error) {
+	//PodSandboxStatus
+	// podSbResponse, err := f.listPodSandBoxResponse()
+	// if err != nil {
+	// 	return nil, err
+	// }
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultPodResourcesTimeout)
+	defer cancel()
+
+	//Pod Resource API client
+	resp, err := f.podResourceClient.List(ctx, &podresourcesapi.ListPodResourcesRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("Can't receive response: %v.Get(_) = _, %v", f.podResourceClient, err)
+	}
+
+	var podResData []PodResources
+
+	for _, podResource := range resp.GetPodResources() {
+		if !f.isWatchable(podResource.GetNamespace()) {
+			log.Printf("SKIP pod %q\n", podResource.Name)
+			continue
+		}
+		// podresource := resp.PodResources[idx]
+    fmt.Printf("podresource: %v", podResource)
+
+
+		podRes := PodResources{
+			Name:      podResource.GetName(),
+			Namespace: podResource.GetNamespace(),
+		}
+		for _, container := range podResource.GetContainers() {
+		contRes := ContainerResources{
+			Name: container.Name,
+		}
+		devs := make(map[string][]string)
+		for _, device := range container.GetDevices() {
+			devs[device.ResourceName] = device.DeviceIds
+		}
+		// cpuList := container.GetCPUs()
+		cpuList := []int32{1,2,3,4,5,6,7,8,9,10}
```

Comment:
hardcoded values? Is this temporary?

Reply:
yes, was temporary when I wasn't using the patched version of kubelet, later moved to obtaining the cpus allocated from getCPU() function